lib/rpc/HealthStateSyncer.ts

```diff
@@ -0,0 +1,130 @@
+import { DynamoDB } from 'aws-sdk'
+import { DocumentClient } from 'aws-sdk/clients/dynamodb'
+import Logger from 'bunyan'
+
+export interface SyncResult {
+  synced: boolean
+  healthScore: number
+}
+
+interface ReadResult {
+  healthScore: number
+  updatedAtInMs: number
+}
+
+export class HealthStateSyncer {
+  private readonly providerId: string
+  private readonly dbTableName: string
+  private ddbClient: DocumentClient
```

Comment:
we can wrap the ddb operations in a repository class to make future porting easier

Reply:
Not sure I understand "in a repository class"... I thought this `HealthStateSyncer` already wraps all DB operations?

Reply:
HealthStateSyncer has some business logic in it, so I was imagining basically a repository class with `get`, `put`, etc. methods. Then HealthStateSyncer becomes agnostic to what db you plug in

Reply:
For a repository class with basic get, put...that sounds like `DynamoDB.DocumentClient`?
Anyway, they are coupled together because I depend on a complex `update` method. The param of this method is deeply coupled with the business logic. That's why it's not very obvious to separate them apart.

Reply:
Yes, I see your point here and I agree. I've refactored my code to separate the health score update logic and storage-backend-specific logic. Please take another look.